crates/ui/src/modals/graphic_picker/fog.rs

```diff
@@ -0,0 +1,409 @@
+                                        .store(ui.ctx(), scroll_area_output.id);
+                                }
+                            }
+                            self.scrolled_on_first_open = true;
+                        }
+                    });
+                });
+
+                egui::TopBottomPanel::top(self.id_source.with("top")).show_inside(ui, |ui| {
+                    ui.add_space(1.0); // pad out the top
+
+                    ui.columns(3, |columns| {
+                        columns[0].horizontal(|ui| {
+                            ui.label("Hue");
+                            if ui.add(egui::Slider::new(fog_hue, 0..=360)).changed() {
+                                if let Some(sprite) = sprite {
+                                    sprite
+                                        .sprite
+                                        .graphic
+                                        .set_hue(&update_state.graphics.render_state, *fog_hue);
+                                }
+                            }
+                        });
+
+                        columns[1].horizontal(|ui| {
+                            ui.label("Opacity");
+                            if ui.add(egui::Slider::new(fog_opacity, 0..=255)).changed() {
+                                if let Some(sprite) = sprite {
+                                    sprite.sprite.graphic.set_opacity(
+                                        &update_state.graphics.render_state,
+                                        *fog_opacity,
+                                    );
+                                }
+                            }
+                        });
+
+                        columns[2].horizontal(|ui| {
+                            ui.label("Blend Type");
+                            ui.add(EnumComboBox::new("blend_type", fog_blend_type));
+                        });
+                    });
+
+                    ui.columns(3, |columns| {
+                        columns[0].horizontal(|ui| {
+                            ui.label("Zoom");
+                            ui.add(egui::DragValue::new(fog_zoom).range(100..=800));
+                        });
+
+                        columns[1].horizontal(|ui| {
+                            ui.label("SX");
+                            ui.add(egui::DragValue::new(fog_sx).range(-256..=256));
+                        });
+
+                        columns[2].horizontal(|ui| {
+                            ui.label("SY");
+                            ui.add(egui::DragValue::new(fog_sy).range(-256..=256));
+                        });
+                    });
+
+                    ui.add_space(1.0); // pad out the bottom
+                });
+                egui::TopBottomPanel::bottom(self.id_source.with("bottom")).show_inside(ui, |ui| {
+                    ui.add_space(ui.style().spacing.item_spacing.y);
+                    crate::components::close_options_ui(ui, &mut keep_open, &mut needs_save);
```

Comment:
The way close_options_ui is used here seems to not work, `Apply` closes the window